pkg/planner/core/operator/logicalop/logical_union_all.go

```diff
@@ -191,7 +193,59 @@ func (p *LogicalUnionAll) ExhaustPhysicalPlans(prop *property.PhysicalProperty)
 
 // CanPushToCop inherits BaseLogicalPlan.LogicalPlan.<21st> implementation.
 
-// ExtractFD inherits BaseLogicalPlan.LogicalPlan.<22nd> implementation.
+// ExtractFD implement base.LogicalPlan.<22nd> interface.
+func (p *LogicalUnionAll) ExtractFD() *fd.FDSet {
+	// basically extract the children's fdSet.
+	childFDs := make([]*fd.FDSet, 0, len(p.children))
+	for _, child := range p.children {
+		childFD := child.ExtractFD()
+		childFDs = append(childFDs, childFD)
+	}
+	// check the output columns' not-null property.
+	res := &fd.FDSet{}
+	notNullCols := intset.NewFastIntSet()
+	for _, col := range p.Schema().Columns {
+		flag := true
+		for _, childFD := range childFDs {
+			if childFD.NotNullCols.Has(int(col.UniqueID)) {
+				flag = false
+				break
+			}
+		}
+		if flag {
+			notNullCols.Insert(int(col.UniqueID))
+		}
+	}
```

Comment:
will it have a better performance if we just perform the intersection among children's intset?

Reply:
Make sense，let me try